doc/doxygen/src/build-system-cpu-conventions.md

```diff
@@ -0,0 +1,177 @@
+$(RIOTCPU)/$(CPU)/models/$(CPU_MODEL)/doc.txt
+~~~~~~~~~~~~~~~~~~~
+
+And then each main `$(CPU)` file will include the `$(CPU_MODEL)` specific one.
+
+This would match the way to do it in RIOT but could create many
+directories and files for maybe no or only few lines.
+
+It could make sense to go with this if the files, especially the
+`Makefile.dep` or `Makefile.include` ones would need many specific definitions
+and handling that are more than variables configuration for the main `CPU`
+files.
+
+
+### `ifneq` based handling ###
+
+To not add any files, the main `CPU` files could add specific definitions if the
+`CPU_MODEL` matches certain values.
+
+~~~~~~~~~~~~~~~~~~~ {.mk}
+ifneq (,$(filter abc1 abd0 ade5,$(CPU_MODEL)))
+  FEATURES_PROVIDED += periph_hwrng
+endif
+~~~~~~~~~~~~~~~~~~~
+
+This allow simple handling when the benefit of having per model definition
+would currently not overcome the overhead of defining it.
+Also, as currently dependencies are not declarative, it is with the previous
+solution one of the only alternative as the following declarative only would
+not work.
+
+An issue with this is that this it not declarative and does not have equivalent
+in a declarative only language that is not Turing complete as `make`.
+But as long as it used with explicit inclusion lists (no wildcard) and no
+blacklists, it is easy to map it to a per model definition from the other
+methods.
+
+
+### Exhaustive per model definition ###
+
+In the case of only declarative values of specific variables, handling different
+`CPU_MODEL` is simply defining one per `CPU_MODEL`.
+When not relying on an `if` evaluation as before, one could define dictionaries
+which are implemented in `make` when using only one file with "namespaced"
+variables names including the `key` and so the `CPU_MODEL` value in the name.
+
+~~~~~~~~~~~~~~~~~~~ {.mk}
+ROM_LEN_stm32f103c8 = 64K
+ROM_LEN_stm32f103cb = 128K
+ROM_LEN_stm32f103rb = 128K
+ROM_LEN_stm32f103re = 512K
+
+ROM_LEN = ROM_LEN_$(CPU_MODEL)
+~~~~~~~~~~~~~~~~~~~
+
+This represents a direct mapping of the `CPU_MODEL` to the variable value.
+This exact format makes sense when there are not too many variables or for
+variables that have a list as value as the following solution does not apply.
+This is true for example for `FEATURES_PROVIDED`.
+
+It is easy to read and to grep, but can give quite indigests lists if all
+`ROM_LEN`, `ROM_ADDR`, `RAM_LEN`, `RAM_ADDR` are all defined for many models.
+
+To not clutter the main `CPU` definition makefiles, these definitions can be
+moved to separate files `models/features.inc.mk` `models/include.inc.mk` or
+similar names. We would still put the `ROM_LEN = ROM_LEN_$(CPU_MODEL)` in the
+main file to separate the "database" from setting the value.
+
+
+### Grouped per model definition ###
+
+Again in the case only delcarative values of specific variables, it can happen
+that many variables are somehow highly correlated.
```

Comment:
In this case variables that are somehow correlated are declared for each `CPU_MODEL`.

Reply:
That changes the style as I wanted to say "in THE case of declarative values that are correlated". In the same way as the section 3 introduction. The `again` was refering to the section 3.
I may not consistent with the style in all solutions now that I re-read them :/

I have your version in a similar way as the last sentence of the paragraph.
 

"somehow highly" is weird indeed :D and I used to much "somehow"

I will fix the typo in `delcarative` as the same time as the rest to not discard your review for nothing.

Reply:
I reworded this, please @fjmolinas take a look